test/Microsoft.Framework.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -41,31 +42,76 @@ public void NonGeneric_GetRequiredServices_ReturnsAllServices()
         }
 
         [Fact]
-        public void GetServicesReturns_A_Service()
+        public void GetRequiredServices_Returns_SingleService()
         {
             // Arrange
-            var serviceProvider = CreateFooServiceProvider(1);
+            var serviceProvider = CreateTestServiceProvider(1);
 
             // Act
             var services = serviceProvider.GetRequiredServices<IFoo>();
 
             // Assert
-            Assert.Contains(services, item => item is Foo1);
-            Assert.Equal(1, services.Count());
+            var item = Assert.Single(services);
+            Assert.IsType<Foo1>(item);
+        }
+
+        [Fact]
+        public void NonGeneric_GetRequiredServices_Returns_SingleService()
+        {
+            // Arrange
+            var serviceProvider = CreateTestServiceProvider(1);
+
+            // Act
+            var services = serviceProvider.GetRequiredServices(typeof(IFoo));
+
+            // Assert
+            var item = Assert.Single(services);
+            Assert.IsType<Foo1>(item);
+        }
+
+        [Fact]
+        public void GetRequiredServices_Returns_CorrectTypes()
+        {
+            // Arrange
+            var serviceProvider = CreateTestServiceProvider(4);
+
+            // Act
+            var services = serviceProvider.GetRequiredServices(typeof(IAnotherFoo));
+
+            // Assert
+            foreach (var service in services)
+            {
+                Assert.True(
+                    service is IAnotherFoo,
+                    $"Service is of type '{service.GetType()}'. Expected: '{typeof(IAnotherFoo)}'.");
+            }
```

Comment:
Used a `foreach` here since the collection is `IEnumerable<object>`, so can't use `Assert.Collection()` here.